### PR TITLE
[mask_rom] Add basic AST initialization support

### PR DESCRIPTION
**Pull request by mundaym**
Initialize the AST by copying its configuration from OTP into the AST registers. A tag value has been added to OTP to allow AST configuration to be skipped if required. The tag value will also mean that AST configuration will be skipped unless OTP has been programmed.

This does not include hardening or any sort of timeout for `ast_init_done`. This will be done once the basic initialization sequence and configuration options have been agreed upon.

There is very limited testing possible with AST currently. The `ast_init_done` signal is hardcoded to 1 in RTL.

### CODE REVIEW

File: sw/device/lib/base/multibits_asm.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_BASE_MULTIBITS_ASM_H_
+#define OPENTITAN_SW_DEVICE_LIB_BASE_MULTIBITS_ASM_H_
+
+/**
+ * Multi-bit boolean values for use in assembly code.
+ *
+ * Please use `multibits.h` instead when writing C code.
```
Comment (mcy) [inline sw/device/lib/base/multibits_asm.h:11]: Do you think we could merge these and condition onf `_ASSEMBLY_` or whatever the define the assembler sets is?

File: sw/device/silicon_creator/mask_rom/mask_rom_start.S [hunk 2]
```diff
@@ -131,6 +135,30 @@ _mask_rom_start_boot:
   li   t0, 0xFFFF0888
   csrc mie, t0
 
+  // Check if AST initialization should be skipped.
+  li   t0, (TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR + \
+            OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET + \
+            OTP_CTRL_PARAM_CREATOR_SW_CFG_AST_INIT_EN_OFFSET)
+  li   t1, MULTIBIT_ASM_BOOL4_TRUE
+  bne  t0, t1, .L_ast_init_skip
```
Comment (zi-v) [inline sw/device/silicon_creator/mask_rom/mask_rom_start.S:143]: I guess that's the part that do the AST configuration skip based on the INIT_EN tag. As far as I understand, it looks fine.
In DEV/PROD* LC states, ROM or ROM_EXT SW should check that ast_init_done was indeed set. At those LC states, INIT_EN is expected to be true. I think that for the engineering samples, it is better to have this checking code in the ROM_EXT.

Comment (alphan) [reply sw/device/silicon_creator/mask_rom/mask_rom_start.S:143]: Created #9685.

**Review by alphan — APPROVED** [review]
@drewmacrae would you mind looking the changes in `BUILD` files? Thanks!

**Review by drewmacrae — APPROVED** [review]
The bazel changes LGTM from a style, linting and functional perspective.

I think depending on the order these get merged in you might see some conflicts with bazel builds and this PR: https://github.com/lowRISC/opentitan/pull/9412 I'll try to remember to check this again if that one gets merged in first.

**Pull request merged**